spec/services/credits/applied_coupon_service_spec.rb

```diff
@@ -97,5 +97,133 @@
+      end
+
+      it 'creates a credit' do
+        result = credit_service.create
+
+        expect(result).to be_success
+
+        expect(result.credit.amount_cents).to eq(25)
+        expect(result.credit.amount_currency).to eq('EUR')
+        expect(result.credit.invoice).to eq(invoice)
+        expect(result.credit.applied_coupon).to eq(applied_coupon)
+      end
+
+      it 'terminates the applied coupon' do
+        result = credit_service.create
+
+        expect(result).to be_success
+        expect(applied_coupon.reload).to be_terminated
+      end
+    end
+
+    context 'when coupon is recurring and fixed amount' do
+      let(:coupon) { create(:coupon, frequency: 'recurring', frequency_duration: 3) }
+
+      let(:applied_coupon) do
+        create(:applied_coupon, coupon: coupon, frequency: 'recurring', frequency_duration: 3, amount_cents: 12)
+      end
+
+      it 'creates a credit' do
+        result = credit_service.create
+
+        expect(result).to be_success
+
+        expect(result.credit.amount_cents).to eq(12)
+        expect(result.credit.amount_currency).to eq('EUR')
+        expect(result.credit.invoice).to eq(invoice)
+        expect(result.credit.applied_coupon).to eq(applied_coupon)
+        expect(result.credit.applied_coupon.frequency_duration).to eq(2)
+      end
+
+      it 'does not terminate the applied coupon' do
+        result = credit_service.create
+
+        expect(result).to be_success
+        expect(applied_coupon.reload).not_to be_terminated
+      end
+
+      context 'and coupon amount is higher than invoice amount' do
+        let(:amount_cents) { 10 }
+
+        it 'limits the credit amount to the invoice amount' do
+          result = credit_service.create
+
+          expect(result).to be_success
+          expect(result.credit.amount_cents).to eq(10)
+        end
+      end
+    end
+
+    context 'when coupon is recurring and percentage' do
+      let(:coupon) { create(:coupon, frequency: 'recurring', frequency_duration: 3, coupon_type: 'percentage') }
+
+      let(:applied_coupon) do
+        create(
+          :applied_coupon,
+          coupon: coupon,
+          frequency: 'recurring',
+          frequency_duration: 3,
+          percentage_rate: 20.00
+        )
+      end
+
+      it 'creates a credit' do
+        result = credit_service.create
+
+        expect(result).to be_success
+
+        expect(result.credit.amount_cents).to eq(25)
+        expect(result.credit.amount_currency).to eq('EUR')
+        expect(result.credit.invoice).to eq(invoice)
+        expect(result.credit.applied_coupon).to eq(applied_coupon)
+        expect(result.credit.applied_coupon.frequency_duration).to eq(2)
+      end
+
+      it 'does not terminate the applied coupon' do
+        result = credit_service.create
+
+        expect(result).to be_success
+        expect(applied_coupon.reload).not_to be_terminated
+      end
+
+      context 'and frequency duration becomes zero' do
```

Comment:
RSpec/ContextWording: Context description should match /^when\b/, /^with\b/, or /^without\b/. (https://rspec.rubystyle.guide/#context-descriptions, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording)